ServerCore/Areas/Identity/Pages/Account/ExternalLogin.cshtml

```diff
@@ -21,22 +21,22 @@
             @* NOTE: If you add more properties here, please add them to Edit.cshtml as well! *@
             <div class="form-group">
                 <label asp-for="Input.Email"></label>
-                <input asp-for="Input.Email" class="form-control" />
+                <input asp-for="Input.Email" class="form-control" maxlength="50" />
```

Comment:
Rather than adding properties to each of the text boxes, it's preferable to add constraints to the backing properties like [StringLength(50)]

Reply:
Will that still prevent characters beyond the limit from being entered to begin with?  The reason I used that method is that a) it doesn't require a database update, and b) it doesn't require users to submit the form before they learn if they've exceeded the max length.

Reply:
Also, I assume that before adding constraints we have to validate that no existing records violate those constraints? Or will the migrator do truncation automatically?

Reply:
Yes, the attributes on backing properties do two things:
1. They trigger built-in validation in both javascript and when the form gets submitted
2. They modify the database to prevent breaking rules there